frontend/src/services/hooks/useGroups.js

```diff
@@ -16,12 +16,9 @@ export const useGroups = () => {
     ) {
       setGroups(['Engineers', 'Scientists', 'DAAdministrators']);
     } else if (process.env.REACT_APP_CUSTOM_AUTH) {
-      if (!auth.user) {
-        dispatch({
-          type: SET_ERROR,
-          error: 'Cannot Set User Groups as the User is not defined'
-        });
-      }
+      // Returning when auth.user is not present
+      // Not dispatching error as useGroups is triggered in auth guard when the user is not authenticated
+      if (!auth.user) return;
```

Comment:
When the user is just authenticated but the auth.user is not yet initialized this dispatch error is thrown on the screen due to code changes here - frontend/src/authentication/components/AuthGuard.js L27 
<img width="1318" alt="image" src="https://github.com/data-dot-all/dataall/assets/71188245/4a67d0c5-2814-4b60-9d5e-468c94062d76">

This would cause unnecessary confusion on the UI and thus I have instead added return

Reply:
Another option would be that I can have a console.log and add a eslint ignore line so that eslint doesn't create problems while building

Reply:
Let me confirm I understood the situation: when we load AuthGuard with or without maintenance enabled, we call useGroups (to get the groups and compare with DAAdmins). In other situations the user is already authenticated when we call useGroups, but in Authguard it might happen that it is not authenticated yet?

Reply:
Yes. That's right. So if the user is not authenticated then this useGroups is called and at that time the `auth.user` is not set so it calls the dispatch showing up on the UI. This occurs not at the login screen but when the user tries to login and then after authentication when the user is sent to the default route (https://github.com/data-dot-all/dataall/blob/bab2f3e6cec36e0de4c9229a25108395910ac2bc/frontend/src/routes.js#L409) there is a fraction of time when this AuthGuard calls the useGroups and at that time the `auth.user` is not set ( as it is still loading the user context after auth ) and this useGroups is triggered , sending a dispatch call which shows up on the UI.